test/inProgressRepositoryService.test.ts

```diff
@@ -2,932 +2,1592 @@ import IAlternative from '@shared/interface/IAlternative';
-      title: inProgressWorkspace.title,
-      description: inProgressWorkspace.therapeuticContext,
-      criteria: expectedCriteria,
-      alternatives: expectedAlternatives,
-      performanceTable: expectedPerformanceTable
-    };
-
-    expect(result).toEqual(expectedResult);
+    });
   });
 
-  it('should throw an error if a cell has neither an effect or a distribution', () => {
-    const inProgressMessage: IInProgressMessage = {
-      workspace: inProgressWorkspace,
-      criteria: criteria,
-      alternatives: alternatives,
-      effects: {
-        ds1Id: {
-          alt1Id: {} as Effect
+  describe('createOrdering', () => {
+    it('should create ordering', () => {
+      const criteria: Record<string, IProblemCriterion> = {
+        crit1Id: {
+          title: 'title',
+          description: 'description',
+          isFavorable: false,
+          dataSources: [
+            {
+              id: dataSource1Id,
+              source: 'reference',
+              unitOfMeasurement: {
+                type: UnitOfMeasurementType.percentage,
+                label: '%'
+              },
+              strengthOfEvidence: 'strengths',
+              uncertainties: 'uncertainties',
+              scale: [0, 100]
+            }
+          ]
+        },
+        crit2Id: {
+          title: 'title',
+          description: 'description',
+          isFavorable: false,
+          dataSources: [
+            {
+              id: 'ds2Id',
+              source: 'reference',
+              unitOfMeasurement: {
+                type: UnitOfMeasurementType.percentage,
+                label: '%'
+              },
+              strengthOfEvidence: 'strengths',
+              uncertainties: 'uncertainties',
+              scale: [0, 100]
+            }
+          ]
         }
-      },
-      distributions: {
-        ds1Id: {
-          alt1Id: {} as Distribution
+      };
+      const alternatives: Record<string, {title: string}> = {
+        alt1Id: {
+          title: 'alternative'
         }
-      }
-    };
-    try {
-      createProblem(inProgressMessage);
-    } catch (error) {
-      expect(error).toBe('Cell without effect and distribution found');
-    }
+      };
+
+      const result = createOrdering(criteria, alternatives);
+
+      const expectedResult: IOrdering = {
+        criteria: ['crit1Id', 'crit2Id'],
+        alternatives: [alternative1Id],
+        dataSources: [dataSource1Id, 'ds2Id']
+      };
+
+      expect(result).toEqual(expectedResult);
+    });
   });
-});
 
-describe('createOrdering', () => {
-  it('should create ordering', () => {
-    const criteria: Record<string, IProblemCriterion> = {
-      crit1Id: {
-        title: 'title',
-        description: 'description',
-        isFavorable: false,
-        dataSources: [
+  describe('buildEmptyInProgress', () => {
+    it('should return a new workspace ready to insert into the database', () => {
+      const result = buildEmptyInProgress();
```

Comment:
we should mock generateUUID and then this whole test is basically one expect call